Add `--tool` flag to specify how to build the crate before flashing.

MabezDev opened the pull request:
Cool project! Has been very useful in removing hacky scripts that invoke esptool.

This PR adds functionality to choose how to build the project, instead of just relying on xargo. I'm attempting to rewrite the xtensa-quickstart repo without needed any other tools to build the code. If it goes well I will be recommending this tool for sure :)

jessebraham commented on cargo-espflash/README.md:
```diff
@@ -2,7 +2,13 @@
 _ESP8266_ and _ESP32_ cross-compiler and serial flasher cargo subcommand.
 
-Currently, `cargo-espflash` requires that you have [xargo](https://github.com/japaric/xargo) installed on your system.
+<!-- Currently, `cargo-espflash` requires that you have [xargo](https://github.com/japaric/xargo) installed on your system. -->
```
This comment can probably be removed.

jessebraham commented on cargo-espflash/README.md:
```diff
@@ -2,7 +2,13 @@
+ - `--tool xargo`, build using `xargo`. Requires [xargo](https://github.com/japaric/xargo) installed on your system.
+ - `--tool xbuild`, build using `cargo xbuild`. Requires [cargo xbuild](https://github.com/rust-osdev/cargo-xbuild) installed on your system.
 
 ## Usage
```
Could you please add `[--tool TOOL]` to the usage example below?

jessebraham commented on cargo-espflash/src/main.rs:
```diff
@@ -68,12 +83,13 @@ struct AppArgs {
 fn usage() -> Result<(), MainError> {
     let mut usage = String::from("Usage: cargo espflash ");
-    usage += "[--ram] [--release] [--example EXAMPLE] ";
+    usage += "[--ram] [--release] [--example EXAMPLE] [--build-tool {{cargo,xargo,xbuild}}]";
```
I believe this should be `[--tool {{cargo,xargo,xbuild}}]`.

jessebraham reviewed (commented):
Thanks for this! I wanted to include this in the original version but ran out of steam 😄 Just a couple of small nitpicks but otherwise looks pretty good to me.

MabezDev commented:
I think I've addressed all your comments, apologies for the delay I've been working through some `build-std` issues (around compiler builtins), acting on that I've chosen to promote `cargo xbuild` to the default build tool until those issues are resolved.

jessebraham commented:
LGTM, thanks! I can't merge in this repo (otherwise I would) so @icewind1991 will have to take care of that.
